### PR TITLE
meson: Use dicts for test/fuzzer definitions

### PR DESCRIPTION
**Pull request by medhefgo**

### CODE REVIEW

**Comment by keszybz** [comment]
mkosi / ci (arch, rolling) is #26172.

File: src/analyze/meson.build [hunk 1]
```diff
@@ -30,11 +30,10 @@ systemd_analyze_sources = files(
...
-               'analyze-verify-util.c',
-               'analyze-verify-util.h'),
-         [libcore,
-          libshared],
-         [],
-         core_includes],
+        { 'sources' : files(
+                'test-verify.c',
+                'analyze-verify-util.c',
+                'analyze-verify-util.h'),
```
Comment (keszybz) [inline src/analyze/meson.build]: What about 4a7ee0a5217ab963c12704dbff428c43d9df81f0? You dropped .h files from the file lists… I think we should do the same here.

File: src/import/meson.build [hunk 2]
```diff
@@ -52,9 +52,8 @@ if conf.get('ENABLE_IMPORTD') == 1
 endif
 
 tests += [
-        [files('test-qcow2.c',
-               'qcow2-util.c'),
-         [],
-         [libz],
-         [], 'HAVE_ZLIB', 'manual'],
+        { 'sources' : files('test-qcow2.c', 'qcow2-util.c'),
```
Comment (keszybz) [inline src/import/meson.build]: No, let's not put multiple files on the same line. We were slowly changing towards a format where all files as listed on separate lines, and if there's more than one file name, a comma and newline after the last file name.

File: src/libudev/meson.build [hunk 3]
```diff
@@ -43,7 +43,6 @@ libudev_pc = custom_target(
 ############################################################
 
 tests += [
-        [files('test-libudev.c'),
-         [libshared,
-          libudev_basic]],
+        { 'sources' : files('test-libudev.c'),
+          'link_with' : [libshared, libudev_basic] },
```
Comment (keszybz) [inline src/libudev/meson.build]: Separate lines please.

**Review by keszybz — CHANGES_REQUESTED** [review]
This is super cool. We support meson>=0.53.2 and dictionaries were already available at that point, so we can do this. But I think we want to keep things on separate lines in general. Otherwise diffs become a mess.

File: src/analyze/meson.build (hunk 1 above)
Comment (medhefgo) [reply src/analyze/meson.build]: Well, regexes don't catch everything :D

**Comment by medhefgo** [comment]
[muon](https://muon.build) has a formatter mode that from a cursory look produces a nicely palatable form. I was playing with the idea of doing a pass on all our meson files (maybe even put a ci pass). Any thoughts on that?

File: src/libudev/meson.build (hunk 3 above)
Comment (medhefgo) [reply src/libudev/meson.build]: for link_with and dependencies? That's gonna be verbose…

**Comment by medhefgo** [comment]
I realized that we have a lot of single-file tests/fuzzers, so I moved those into their own list. This way, the more verbose test definitions are much more palatable.

I also tried a way to share some properties with a new `base` property. Alternatively, this sharing could be achieved in a similar way to how it's done with `common_test_dependencies` (aka instead of merging things in the forearch it's done manually in the test definitions).

PTAL

**Review by yuwata — APPROVED** [review]
Nice! LGTM.

**Pull request merged**